lib/validators/options_processor/options/no_empty.js

```diff
@@ -25,10 +27,15 @@ class NoEmpty {
 
     if (typeof input === 'string') {
       isMergeable = input.trim().length !== 0
-    } else if (Array.isArray(input)) {
-      isMergeable = input.length !== 0
+    } else if (Array.isArray(input) && input[0] != null) {
```

Comment:
I don't understand the changes in these two lines. can you help me understand what you are trying to achieve here?

Reply:
@shine2lay I'm fixing these lines of tests I've added
```
verify(true, null, [null], 'fail')
verify(true, undefined, [undefined], 'fail')
```

Reply:
but why are those tests needed? when it's an array `no_empty` we should just check if the array is not empty i.e `array.length !== 0` . why are we checking for null/undefined value there?

Reply:
can you describe a use case where that would be useful?

Reply:
@tenshiAMD can you please address this comment? I am still not sure why you added these tests where `[null]` is a use case